var/spack/repos/builtin/packages/activeharmony/package.py

```diff
@@ -6,14 +6,31 @@
 from spack import *
 
 
-class Activeharmony(Package):
+class Activeharmony(MakefilePackage):
     """Active Harmony: a framework for auto-tuning (the automated search for
-       values to improve the performance of a target application)."""
+    values to improve the performance of a target application)."""
+
     homepage = "http://www.dyninst.org/harmony"
     url      = "http://www.dyninst.org/sites/default/files/downloads/harmony/ah-4.5.tar.gz"
 
-    version('4.5', sha256='31d9990c8dd36724d336707d260aa4d976e11eaa899c4c7cc11f80a56cdac684')
+    version('4.6.0', sha256='9ce5009cfd8e2f4cf5f3536e1fea9993414fc25920fc90d0a2cb56f044787dbb')
+    version('4.5',   sha256='31d9990c8dd36724d336707d260aa4d976e11eaa899c4c7cc11f80a56cdac684')
 
-    def install(self, spec, prefix):
+    patch('fix_logical_bug_in_slave_list_parsing.patch', sha256='3e000616f84de80b262efcae7559d65eed0efcd53e915580dab63b0ffbbb8bf2', when='@4.6.0')
+
+    def build(self, spec, prefix):
         make("CFLAGS=-O3")
+
+    @when('@:4.5')
+    def install(self, spec, prefix):
         make("install", 'PREFIX=%s' % prefix)
+
+    @when('@4.6.0:')
+    def install(self, spec, prefix):
+        make("install")
```

Comment:
How does this version know where to install to? Just it just install to the current directory?

Reply:
Yes, it remove `PREFIX` variable in `Makefile` and only install it in the current directory.